### ISSUE TITLE
iOS 13 navigation bar appearance needs update

### ISSUE THREAD

**Issue by Drusy** [stripe/stripe-ios#1391]
## Summary
iOS 13 include a new navigation bar appearance management via `UINavigationBarAppearance`. 
The navigation bar is now totally transparent (cf. screenshot at the end of the issue)
It is disappointing but it seems to impact apps currently on the App Store and compiled with Xcode 10.3

## Code to reproduce
`paymentContext.presentPaymentOptionsViewController()`

## iOS version
iOS 13 only

## Installation method / SDK version
Carthage 
> github "stripe/stripe-ios" "v17.0.2"

## Other information
![Simulator Screen Shot - iPhone SE - 2019-09-26 at 16 24 44](https://user-images.githubusercontent.com/2833240/65697291-f9050b00-e07a-11e9-9e69-1a5dfad18ab9.png)

**Comment by davidme-stripe** [comment]
Do you still see this after setting `paymentContext!.theme.translucentNavigationBar = true`? It should be set to that by default in SDK 17.0.2, but it’s possible that it was unset somehow.

**Comment by Drusy** [comment]
Yes, having `translucentNavigationBar` to `true` shows the above screenshot. Setting it to `false` shows a totally black navigation bar as referenced here #1369 
Except, I don't run the beta version of iOS, I am using iOS 13.1 official release

**Comment by davidme-stripe** [comment]
Are you building your app with Xcode 10.2? You’ll need to compile with Xcode 11 to get the correct behavior on iOS 13.

If it’s still transparent, try setting `paymentContext.theme.secondaryBackgroundColor`, it may have been unset somehow.

**Comment by Drusy** [comment]
Tried both Xcode 10.3 and Xcode 11, same result :(

```swift
theme.primaryBackgroundColor = .blue
theme.secondaryBackgroundColor = .red
```

Results in this, unfortunately we still get a transparent navigation bar.
The solution is probably using `UINavigationBarAppearance` and `navigationBar.standardAppearance`, `navigationBar.scrollEdgeAppearance`, `navigationBar.compactAppearance`

![Simulator Screen Shot - iPhone SE - 2019-09-26 at 16 57 02](https://user-images.githubusercontent.com/2833240/65699723-ca892f00-e07e-11e9-84bb-0459549d5aef.png)

![Simulator Screen Shot - iPhone SE - 2019-09-26 at 16 57 17](https://user-images.githubusercontent.com/2833240/65699728-cbba5c00-e07e-11e9-927d-d9598bd9cd3d.png)

**Comment by davidme-stripe** [comment]
We are setting the `scrollEdgeAppearance` and `compactAppearance`, but only when building with the iOS 13 SDK.

Two other ideas, because you're using Carthage:
1) Carthage may be building Stripe.framework using Xcode 10.2 if you still have it installed. On your system, `xcode-select -p` should return the path to Xcode 11. If it doesn't, you'll want to `sudo xcode-select --switch $YOUR_PATH_TO_XCODE_11`. 
2) The binary framework on our GitHub releases page uses Xcode 10.2 for backwards compatibility reasons, and it's possible that Carthage has picked up that pre-compiled framework. The fix for this is on our end — we may have to start distributing an Xcode 11 version instead.

**Comment by Drusy** [comment]
Yes ! I still have the terminal history, I did use Xcode 11 to compile using Carthage but it used the precompiled binary on GitHub:
> *** Downloading stripe-ios.framework binary at "Version 17.0.2"

I will now compile Stripe using `--no-use-binaries`:
> carthage update stripe-ios --use-submodules --platform iOS --no-use-binaries

And it now works, thanks for your time, and I hope this thread will help someone else.
Will you build using Xcode 11 for the future release or should we keep the `--no-use-binaries` option ?

![Simulator Screen Shot - iPhone SE - 2019-09-26 at 19 25 17](https://user-images.githubusercontent.com/2833240/65710497-6e7cd580-e093-11e9-968f-f80ae329cad4.png)

**Comment by davidme-stripe** [comment]
This is on us — we're going to replace the precompiled binary with an Xcode 11 version, and we'll ask folks using Xcode 10.2 to build it themselves using the `--no-use-binaries` flag. Thanks for your effort in helping us track this down!

**Comment by davidme-stripe** [comment]
We've replaced the binary, and we'll make sure future builds are built with Xcode 11 in #1392.

**Comment by aprashar3** [comment]
In your app did finish launching with options methods add the following lines of code. Change the colour property as required. It worked for me hope it work for others also.

Obj-c

if (@available(iOS 13.0, *)) {
        UINavigationBarAppearance *appearance = [[UINavigationBarAppearance alloc] init];
        [appearance configureWithOpaqueBackground];
        appearance.backgroundColor = UIColor.whiteColor;
        [appearance setTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor blackColor]}];
        [appearance setLargeTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor blackColor]}];
        UINavigationBar.appearance.scrollEdgeAppearance = appearance;
        UINavigationBar.appearance.standardAppearance = appearance;
    }

Swift

let coloredAppearance = UINavigationBarAppearance()
coloredAppearance.configureWithOpaqueBackground()
coloredAppearance.backgroundColor = .white
coloredAppearance.titleTextAttributes = [.foregroundColor: UIColor.black]
coloredAppearance.largeTitleTextAttributes = [.foregroundColor: UIColor.black]
       
UINavigationBar.appearance().standardAppearance = coloredAppearance
UINavigationBar.appearance().scrollEdgeAppearance = coloredAppearance

Found a great link regarding the same : https://sarunw.com/posts/uinavigationbar-changes-in-ios13/
